### PR TITLE
fix climate

### PR DESCRIPTION
**Pull request by marcelveldt**
fix multi target setpoints

### CODE REVIEW

**Comment by marcelveldt** [comment]
@MartinHjelmare feel free to merge if you do not spot any issues

File: custom_components/zwave_mqtt/climate.py [hunk 1]
```diff
@@ -460,6 +489,6 @@ def _mode(self) -> None:
 
     def _current_mode_setpoints(self) -> Tuple:
         """Return a tuple of current setpoint Z-Wave value(s)."""
-        current_mode = str(self.values.primary.value).lower()
+        current_mode = self.values.primary.value["Selected_id"]
```
Comment (quinnhosler) [inline custom_components/zwave_mqtt/climate.py:492]: ``VALUE_SELECTED`` already exists as a constant for ``Selected_id``

Comment (marcelveldt) [reply custom_components/zwave_mqtt/climate.py:492]: thanks!
BTW: I assume you own the other type of thermostat yourself ?
This "multi setpoint" type of thermostat was broken...

Comment (quinnhosler) [reply custom_components/zwave_mqtt/climate.py:492]: I've never used the original ZW component and didn't know they had merged heating and cooling together. I thought the way it was working for me was the expected behavior with the separate entities

Comment (marcelveldt) [reply custom_components/zwave_mqtt/climate.py:492]: No problem, this PR will fix it.

File: custom_components/zwave_mqtt/climate.py [hunk 2]
```diff
@@ -40,23 +41,47 @@
 
 ATTR_FAN_ACTION = "fan_action"
 
+
+class ThermostatMode(IntEnum):
```
Comment (MartinHjelmare) [inline custom_components/zwave_mqtt/climate.py:45]: We should move this to the library.

File: custom_components/zwave_mqtt/climate.py [hunk 3]
```diff
@@ -40,23 +41,47 @@
...
+    DRY = 8
+    MOIST = 9
+    AUTO_CHANGE_OVER = 10
+    HEATING_ECON = 11
+    COOLING_ECON = 12
+    AWAY = 13
+    FULL_POWER = 14
+
+
 MODE_SETPOINT_MAPPINGS = {
```
Comment (MartinHjelmare) [inline custom_components/zwave_mqtt/climate.py:66]: Does this also belong in the library?

File: custom_components/zwave_mqtt/climate.py [hunk 4]
```diff
@@ -40,23 +41,47 @@
...
+    ThermostatMode.COOLING_ECON: ("setpoint_eco_cooling",),
+    ThermostatMode.AWAY: ("setpoint_away_heating", "setpoint_away_cooling"),
+    ThermostatMode.FULL_POWER: ("setpoint_full_power",),
 }
 
+
 # strings, OZW and/or qt-ozw does not send numeric values
+# https://github.com/OpenZWave/open-zwave/blob/master/cpp/src/command_classes/ThermostatOperatingState.cpp
 HVAC_CURRENT_MAPPINGS = {
     "idle": CURRENT_HVAC_IDLE,
```
Comment (MartinHjelmare) [inline custom_components/zwave_mqtt/climate.py:86]: Would be good with constants/enums in the library for the left part of this dict.

File: custom_components/zwave_mqtt/climate.py [hunk 5]
```diff
@@ -80,6 +105,7 @@
     HVAC_MODE_AUTO,
 ]
 
+# Map Z-Wave HVAC Mode to Home Assistant value
 ZW_HVAC_MODE_MAPPINGS = {
```
Comment (MartinHjelmare) [inline custom_components/zwave_mqtt/climate.py:109]: Here too.

**Review by MartinHjelmare — COMMENTED** [review]
Just commenting on the constants that we probably want in the library.

File: custom_components/zwave_mqtt/climate.py (hunk 2 above)
Comment (marcelveldt) [reply custom_components/zwave_mqtt/climate.py:45]: Yeah we need a close look at all files what can be turned into more enums/constants in the lib.

**Pull request merged**